pkg/k8shandler/indexmanagement/indexmanagement_test.go

```diff
@@ -27,6 +27,104 @@ var _ = Describe("Indexmanagement", func() {
 		}
 	})
 
+	Describe("IndexManagemet Policy with 0 minutes, hours, days, weeks, months", func() {
+		It("should correctly parse 0m", func() {
+			retentionPolicy = &logging.RetentionPoliciesSpec{
+				App: &logging.RetentionPolicySpec{
+					MaxAge: esapi.TimeUnit("0m"),
+				},
+				Infra: &logging.RetentionPolicySpec{
+					MaxAge: esapi.TimeUnit("0m"),
+				},
+				Audit: &logging.RetentionPolicySpec{
+					MaxAge: esapi.TimeUnit("0m"),
+				},
+			}
+			spec := NewSpec(retentionPolicy)
+			Expect(len(spec.Policies)).To(Equal(3))
+			Expect(len(spec.Mappings)).To(Equal(3))
+			Expect(spec.Policies[0].Phases.Delete.MinAge).To(Equal(esapi.TimeUnit("0m")))
+			Expect(spec.Policies[1].Phases.Delete.MinAge).To(Equal(esapi.TimeUnit("0m")))
+			Expect(spec.Policies[2].Phases.Delete.MinAge).To(Equal(esapi.TimeUnit("0m")))
+		})
+		It("should correctly parse 0h", func() {
+			retentionPolicy = &logging.RetentionPoliciesSpec{
+				App: &logging.RetentionPolicySpec{
+					MaxAge: esapi.TimeUnit("0h"),
+				},
+				Infra: &logging.RetentionPolicySpec{
+					MaxAge: esapi.TimeUnit("0h"),
+				},
+				Audit: &logging.RetentionPolicySpec{
+					MaxAge: esapi.TimeUnit("0h"),
+				},
+			}
+			spec := NewSpec(retentionPolicy)
+			Expect(len(spec.Policies)).To(Equal(3))
+			Expect(len(spec.Mappings)).To(Equal(3))
+			Expect(spec.Policies[0].Phases.Delete.MinAge).To(Equal(esapi.TimeUnit("0h")))
```

Comment:
@ewolinetz the original intent why this bug was exposed was @RiRa12621 was trying to disable collection of a specific log source (I think).  Implication is never delete?  This corrects the parsing error but I think it will mean: delete the indices everytime cron runs because the indices will always be greater then the minage.  Should we restrict you to defining something > 0 or is it sufficient to say "customer told me continually delete so thats what we will do"?

Reply:
agreed, this currently will mean delete every time it will run. if we want it to mean "don't delete" (which, I personally don't think we should offer because it means the cluster will fall over) we should update the logic in the EO cron scripts first so we don't have a gap where we constantly delete logs.